tests/common/devices/vmhost.py

```diff
@@ -0,0 +1,21 @@
+from tests.common.devices.base import AnsibleHostBase
+
+
+class VMHost(AnsibleHostBase):
+    """
+    @summary: Class for VM server
+
+    For running ansible module on VM server
+    """
+
+    def __init__(self, ansible_adhoc, hostname):
+        AnsibleHostBase.__init__(self, ansible_adhoc, hostname)
+
+    @property
+    def external_port(self):
+        if not hasattr(self, "_external_port"):
+            vm = self.host.options["variable_manager"]
+            im = self.host.options["inventory_manager"]
+            hostvars = vm.get_vars(host=im.get_host(self.hostname), include_delegate_to=False)
```

Comment:
Why not use the `get_test_server_vars` function in `tests.common.utilities.py` for getting `external_port`?

Reply:
Looks like `get_test_server_vars` only load the host variable stored in the inventory instead of the variable files under `ansible/host_vars`.

Reply:
Can you add a utility function like `get_test_server_visible_vars`?

Reply:
Yes, I will have another PR for this and will modify this part then, is it ok?